mkosi/initrd.py

```diff
@@ -88,89 +89,102 @@ def main() -> None:
+            if Path(d).exists():
+                cmdline += ["--include", d]
+
+        with tempfile.TemporaryDirectory() as d:
+            # Make sure we don't use any of mkosi's default repositories.
+            for p in (
+                "yum.repos.d/mkosi.repo",
+                "apt/sources.list.d/mkosi.sources",
+                "zypp/repos.d/mkosi.repo",
+                "pacman.conf",
+            ):
+                (Path(d) / "etc" / p).parent.mkdir(parents=True, exist_ok=True)
+                (Path(d) / "etc" / p).touch()
+
+            # Copy in the host's package manager configuration.
+            for p in (
+                "dnf",
+                "yum.repos.d/",
+                "apt",
+                "zypp",
+                "pacman.conf",
+                "pacman.d/",
+            ):
+                if not (Path("/etc") / p).exists():
+                    continue
+
+                (Path(d) / "etc" / p).parent.mkdir(parents=True, exist_ok=True)
+                if (Path("/etc") / p).resolve().is_file():
+                    shutil.copy2(Path("/etc") / p, Path(d) / "etc" / p)
+                else:
+                    shutil.copytree(
+                        Path("/etc") / p,
+                        Path(d) / "etc" / p,
+                        ignore=shutil.ignore_patterns("gnupg"),
+                        dirs_exist_ok=True,
+                    )
+
+            cmdline += ["--sandbox-tree", d]
+
+            # Prefer dnf as dnf5 has not yet officially replaced it and there's a much bigger chance that
+            # there will be a populated dnf cache directory.
+            run(
+                cmdline,
+                stdin=sys.stdin,
+                stdout=sys.stdout,
+                env={"MKOSI_DNF": dnf.name} if (dnf := find_binary("dnf", "dnf5")) else {},
+            )
+
+            if args.output_dir:
+                Path(args.output_dir).mkdir(parents=True, exist_ok=True)
             else:
-                shutil.copytree(
-                    Path("/etc") / p,
-                    Path(d) / "etc" / p,
-                    ignore=shutil.ignore_patterns("gnupg"),
-                    dirs_exist_ok=True,
-                )
-
-        cmdline += ["--sandbox-tree", d]
-
-        # Prefer dnf as dnf5 has not yet officially replaced it and there's a much bigger chance that there
-        # will be a populated dnf cache directory.
-        run(
-            cmdline,
-            stdin=sys.stdin,
-            stdout=sys.stdout,
-            env={"MKOSI_DNF": dnf.name} if (dnf := find_binary("dnf", "dnf5")) else {},
-        )
+                args.output_dir = Path.cwd()
+
+            log_notice(f"Copying {staging_dir}/{args.output} to {args.output_dir}/{args.output}")
+            # mkosi symlinks the expected output image, so dereference it
+            copy_tree(
+                Path(f"{staging_dir}/{args.output}").resolve(),
+                Path(f"{args.output_dir}/{args.output}"),
```

Comment:
This will rename the file when copying. I don't think we want that, we want the full name with extension and compression. Just specifying the output directory here will be sufficient

Reply:
Hmm.. I don't think we want the full name here, if someone wants to produce an image called `initrd-<kver>`, he wants that name and not e.g. `initrd-<kver>.cpio.zst`

Reply:
Makes sense, let's leave it like this

Reply:
Ok, done.